s3-proxy/Dockerfile

```
@@ -14,12 +14,15 @@ RUN set -x \
     && addgroup --system nginx \
     && adduser --system --disabled-login --ingroup nginx --no-create-home --home /nonexistent --shell /bin/false nginx
 
+COPY s3.patch /root/s3.patch
+
 RUN set -x \
     && apt-get update \
-    && apt-get install -y gettext \
+    && apt-get install -y gettext patch \
     && apt-get install -y curl gcc make libssl-dev libpcre3-dev zlib1g-dev \
     && curl -L http://nginx.org/download/nginx-${NGINX_VERSION}.tar.gz | tar zx \
     && cd nginx-${NGINX_VERSION} \
+    && patch -p1 < /root/s3.patch \
```

minor: with this you don't need an further comments
```suggestion
    && patch -p1 < /root/s3_encode_plus.patch \
```